src/back/glsl.rs

```diff
@@ -1632,3 +1663,191 @@ fn write_format_glsl(format: StorageFormat) -> &'static str {
         StorageFormat::Rgba32Float => "rgba32f",
     }
 }
+
+fn collect_texture_mapping(
+    module: &Module,
+    functions: &FastHashMap<Handle<Function>, String>,
+) -> Result<Vec<TextureMapping>, Error> {
+    fn collect_texture_mapping_expr(
+        func: &Function,
+        expr: Handle<Expression>,
+        mappings: &mut Vec<TextureMapping>,
+    ) -> Result<(), Error> {
+        match func.expressions[expr] {
+            Expression::Access { base, index } => {
+                collect_texture_mapping_expr(func, base, mappings)?;
```

Comment:
We have the logic of walking through expressions in `proc::Interface`. It would be nice to share this code wider by separating the visitor logic from the traversal.